src/test/java/cloud/fogbow/ras/core/OrderControllerTest.java

```diff
@@ -167,36 +146,271 @@ public void testGetInactiveOrder() throws InstanceNotFoundException {
     @Test(expected = InstanceNotFoundException.class)
     public void testGetInvalidOrder() throws FogbowException {
         // exercise
-        this.ordersController.getOrder("invalid-order-id");
+        this.ordersController.getOrder(INVALID_ORDER_ID);
+    }
+
+    //test case: Checks if attempt to activate null order throws UnexpectedException
+    @Test(expected = UnexpectedException.class)
+    public void testActivateNullOrder() throws FogbowException {
+        // set up
+        Order order = null;
+
+        // exercise
+        this.ordersController.activateOrder(order);
+    }
+
+    // test case: Attempt to activate the same order more than
+    // once must throw UnexpectedException
+    @Test(expected = UnexpectedException.class)
+    public void testActivateOrderTwice() throws FogbowException {
+        // set up
+        Order order = createLocalOrder(getLocalMemberId());
+
+        // exercise
+        this.ordersController.activateOrder(order);
+        this.ordersController.activateOrder(order);
+    }
+
+    // test case: Attempt to deactivate the same order more than
+    // once must throw UnexpectedException
+    @Test(expected = UnexpectedException.class)
+    public void testDeactivateOrderTwice() throws FogbowException {
+        // set up
+        Order order = createLocalOrder(getLocalMemberId());
+
+        // exercise
+        this.ordersController.activateOrder(order);
+        this.ordersController.deactivateOrder(order);
+        this.ordersController.deactivateOrder(order);
+    }
+
+    // test case: Checks if deactivateOrder changes it's state to DEACTIVATED
+    @Test
+    public void testDeactivateOrderSuccess() throws FogbowException {
+        // set up
+        Order order = createLocalOrder(getLocalMemberId());
+
+        // exercise
+        this.ordersController.activateOrder(order);
+        this.ordersController.deactivateOrder(order);
+
+        // verify
+        Assert.assertEquals(order.getOrderState(), OrderState.DEACTIVATED);
+    }
+
+    // test case: Creates an order with dependencies and check if the order id
+    // will be inserted into dependencies.
+    @Test
+    public void testCreateOrderWithDependencies() throws FogbowException {
+        // set up
+        ComputeOrder computeOrder = createLocalComputeOrder();
+        VolumeOrder volumeOrder1 = createLocalVolumeOrder();
+        VolumeOrder volumeOrder2 = createLocalVolumeOrder();
+
+        this.ordersController.activateOrder(computeOrder);
+        this.ordersController.activateOrder(volumeOrder1);
+        this.ordersController.activateOrder(volumeOrder2);
+
+        AttachmentOrder attachmentOrder1 = createLocalAttachmentOrder(computeOrder, volumeOrder1);
+        AttachmentOrder attachmentOrder2 = createLocalAttachmentOrder(computeOrder, volumeOrder2);
+
+        // exercise
+        this.ordersController.activateOrder(attachmentOrder1);
+        this.ordersController.activateOrder(attachmentOrder2);
+
+        // verify
```

Comment:
The checkOrderDependencies in the OrderController class thrown an exception instead of using boolean values. I think we should take the opportunity to refactor this code

Reply:
I'll refactor this method and fix the test, as also remove the orderHasDependencies method from the test class.